### PR TITLE
Make login

### PR DESCRIPTION
**Pull request by ZakHolliman**
Creating a pull request to figure out what I need to finish up in order to get the Login page merged into master before moving on to User or Preference Page

### CODE REVIEW

File: src/http.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import axios from "axios";
+// import { useAuthStore } from './stores/auth'
+
+// const store = useAuthStore();
+
+export default function () {
+  return axios.create({
+    baseURL: "http://localhost:3005",
```
Comment (SpencerLefever) [inline src/http.ts:8]: Make base url an env variable

File: src/stores/auth.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+      persistence: useLocalStorage("auth", {
+        token: "",
+        showLogin: true,
+      }),
+    };
+  },
+  actions: {
+    login() {
+      http()
+        .post("/api/v1/auth/login", {
```
Comment (SpencerLefever) [inline src/stores/auth.ts:22]: this could also be an env variable

File: src/stores/auth.ts [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+          uid: this.nonpersistence.uid,
+          password: this.nonpersistence.password,
+        })
+        .then((response) => {
+          this.persistence.token = response.data.token;
+          this.persistence.showLogin = false;
+          // Clean up debug
+        })
+        .catch(function (error) {
+          console.log(error);
```
Comment (SpencerLefever) [inline src/stores/auth.ts:32]: remove console.log from code before merging to production branch

**Review by SpencerLefever — CHANGES_REQUESTED** [review]

**Review by B-Ryan4 — COMMENTED** [review]
Code looks good from what I can tell. I would recommend adding comments to your code so if others need to look at it they can see what is happening

**Comment by B-Ryan4** [comment]
looks goos should be ready to go

File: package.json [hunk 4]
```diff
@@ -10,7 +10,9 @@
   },
   "dependencies": {
     "@quasar/extras": "^1.12.5",
+    "@vueuse/core": "^8.4.2",
     "axios": "^0.27.2",
+    "dotenv": "^16.0.2",
```
Comment (wrathofrathma) [inline package.json:15]: we ended up not using dotenv on the frontend i think

**Review by wrathofrathma — APPROVED** [review]
Requesting for review is fine, but also don't forget to use the project board to mark what stage it's in. That's what I check for.

Why is there a random empty store file called "response.json"?

**Pull request merged**